ops/op2/signature.rs

```diff
@@ -488,10 +577,12 @@ fn parse_generics(
 }
 
 fn parse_attributes(attributes: &[Attribute]) -> Result<Attributes, ArgError> {
-  let attrs = attributes
-    .iter()
-    .filter_map(parse_attribute)
-    .collect::<Vec<_>>();
+  let mut attrs = vec![];
```

Comment:
**suggestion:** Maybe use `Vec::with_capacity(attrs.len())`?

Reply:
Good point. As this code is leaning more towards readability than performance, I'm going to punt on this for now -- if we find that op parsing is slow overall, it may require a bit of perf work over the whole codebase.